packages/chip/src/utils/getTruncatedName/getTruncatedName.spec.tsx

```diff
@@ -0,0 +1,197 @@
+
+import { getTruncatedName } from '.';
+
+const stringToTruncate = 'hey';
+
+describe('packages/chip/utils/getTruncatedName', () => {
+  describe('renders correct label when the chipCharacterLimit is `2`', () => {
+    test('and the chipTruncationLocation is `End`', () => {
+      const results = getTruncatedName(
+        2,
+        TruncationLocation.End,
+        stringToTruncate,
+      );
+      expect(results).toEqual('…');
+    });
+
+    test('and the chipTruncationLocation is `Start`', () => {
+      const results = getTruncatedName(
+        2,
+        TruncationLocation.Start,
+        stringToTruncate,
+      );
+      expect(results).toEqual('…');
+    });
+
+    test('and the chipTruncationLocation is `Middle`', () => {
+      const results = getTruncatedName(
+        2,
+        TruncationLocation.Middle,
+        stringToTruncate,
+      );
+      expect(results).toEqual('…');
+    });
+
+    test('and the chipTruncationLocation is `None`', () => {
+      const results = getTruncatedName(
+        2,
+        TruncationLocation.None,
+        stringToTruncate,
+      );
+      expect(results).toEqual('hey');
+    });
+  });
+
+  describe('renders correct label when the chipCharacterLimit is `3`', () => {
+    test('and the chipTruncationLocation is `End`', () => {
+      const results = getTruncatedName(
+        3,
+        TruncationLocation.End,
+        stringToTruncate,
+      );
+      expect(results).toEqual('…');
+    });
+
+    test('and the chipTruncationLocation is `Start`', () => {
+      const results = getTruncatedName(
+        3,
+        TruncationLocation.Start,
+        stringToTruncate,
+      );
+      expect(results).toEqual('…');
+    });
+
+    test('and the chipTruncationLocation is `Middle`', () => {
+      const results = getTruncatedName(
+        3,
+        TruncationLocation.Middle,
+        stringToTruncate,
+      );
+      expect(results).toEqual('…');
+    });
+
+    test('and the chipTruncationLocation is `None`', () => {
+      const results = getTruncatedName(
+        3,
+        TruncationLocation.None,
+        stringToTruncate,
+      );
+      expect(results).toEqual('hey');
+    });
+  });
+
+  describe('renders correct label when the chipCharacterLimit is `4`', () => {
+    test('and the chipTruncationLocation is `End`', () => {
+      const results = getTruncatedName(
+        4,
+        TruncationLocation.End,
+        stringToTruncate,
+      );
+      expect(results).toEqual('h…');
+    });
+
+    test('and the chipTruncationLocation is `Start`', () => {
+      const results = getTruncatedName(
+        4,
+        TruncationLocation.Start,
+        stringToTruncate,
+      );
+      expect(results).toEqual('…y');
+    });
+
+    test('and the chipTruncationLocation is `Middle`', () => {
+      const results = getTruncatedName(
+        4,
+        TruncationLocation.Middle,
+        stringToTruncate,
+      );
+      expect(results).toEqual('…y');
+    });
```

Comment:
Looks like a conscious decision to prefer 'End' only one end of 'Middle' is visible. Is this worth noting down? I guess people would just realize when they use it and won't really care?

Reply:
I don't think we need to go too into the weeds of how middle should work seeing as this test case would never even be an issue inside Chip since we have this check: `(label as string).length > (chipCharacterLimit as number);`. 

Middle is trying to halve the word and add in the ellipsis based on the limit but in this test case the limit is 4 and the label is only 3 letters so the bulk of the truncated word will be the 3 dots and the one left over character.